features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/sha/sha_alt_hw.c

```diff
@@ -102,12 +91,11 @@ void mbedtls_sha1_hw_finish(crypto_sha_context *ctx, unsigned char output[20])
         crypto_sha_update_nobuf(ctx, ctx->buffer, ctx->buffer_left, 1);
         ctx->buffer_left = 0;
         crypto_sha_getinternstate(output, 20);
-    
+
         CRPT->HMAC_CTL |= CRPT_HMAC_CTL_STOP_Msk;
-    }
-    else {
+    } else {
         mbedtls_sha1_sw_context ctx_sw;
-    
+
```

Comment:
The H/W isn't stopped in case of `ctx->total == 0`, and also not if `mbedtls_sha1_hw_finish` is never called. Looking at your comment in `mbedtls_sha1_hw_starts`, it appears unproblematic to issue `STOP` signals even if the device has already stopped, so I'd suggest to move the `CRPT->SHA_CTL |= CRPT_SHA_CTL_STOP_Msk` from `mbedtls_sha1_hw_finish` to the end of the function and to add the same to `mbedtls_sha1_hw_free`.

Reply:
@hanno-arm I fix it in https://github.com/ARMmbed/mbed-os/pull/4925/commits/4618cb72580956ad8cf97ac1513ac7a8b125b92c.